pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -462,55 +547,17 @@ func doInstall(ctx *zedmanagerContext, uuidStr string,
 		log.Infof("StorageStatus Name: %s, safename %s, ImageSha256: %s",
 			ss.Name, safename, ss.ImageSha256)
 
-		// Shortcut if image is already verified
-		vs := lookupVerifyImageStatusAny(ctx, safename, ss.ImageSha256)
-		// Handle post-reboot verification in progress by allowing
-		// types.DOWNLOADED. If the verification later fails we will
-		// get a delete of the VerifyImageStatus and skip this
-		if vs != nil && (vs.State == types.DELIVERED || vs.State == types.DOWNLOADED) {
-			switch vs.State {
-			case types.DELIVERED:
-				log.Infof("doInstall found verified image for %s sha %s\n",
-					safename, ss.ImageSha256)
-
-			case types.DOWNLOADED:
-				log.Infof("doInstall found downloaded/verified image for %s sha %s\n",
-					safename, ss.ImageSha256)
-			}
-			if vs.Safename != safename {
-				// If found based on sha256
-				log.Infof("doInstall found diff safename %s\n",
-					vs.Safename)
-			}
-			if ss.IsContainer {
-				log.Debugf("doInstall: Container. ss.ContainerImageID: %s, "+
-					"vs.IsContainer = %t, vs.ContainerImageID: %s\n",
-					ss.ContainerImageID, vs.IsContainer, vs.ContainerImageID)
-				if len(ss.ContainerImageID) == 0 {
-					ss.ContainerImageID = vs.ContainerImageID
-					status.ContainerImageID = vs.ContainerImageID
-					changed = true
-				}
-			}
-			// If we don't already have a RefCount add one
-			if !ss.HasVerifierRef {
-				log.Infof("doInstall !HasVerifierRef vs. RefCount %d for %s\n",
-					vs.RefCount, vs.Safename)
-				// We don't need certs since Status already exists
-				MaybeAddVerifyImageConfig(ctx, uuidStr, vs.Safename, *ss, false)
-				ss.HasVerifierRef = true
-				changed = true
-			}
+		// Check if VerifierStatus already exists.
+		vs, statusUpdated := doInstallProcessStorageEntriesWithVerifiedImage(
+			ctx, status, ss)
+		if vs != nil {
+			log.Debugf("doInstall: Verified image exists. StorageStatus "+
+				"Name: %s, safename %s, ImageSha256: %s",
+				ss.Name, safename, ss.ImageSha256)
+			changed = changed || statusUpdated
```

Comment:
Shouldn't you update changed even if vs is nil? You shouldn't assume when doInstallProcessStorage... updates status, since that breaks modularity.

Reply:
Ah.. I see what u r saying.. The intention is that statusUpdated can be true only when vs != nil.. 
Hmm.. What u say is actually better.. We are passing ssPtr..So we don't need such assumptions. That would keep it simple.. Fixed..